### PR TITLE
 fix(DB/Quest): Missing RP event after completing the quest "Deciphering the Book"

### PR DESCRIPTION
**Pull request by Tony931023**
credits to https://github.com/unified-db/Database/blob/17f8f6425a4835591606930ab94694d85fd707c0/Updates/452_quest_9557.sql and ty @Grz3s
Co-Authored-By: Rushor <Rushor@users.noreply.github.com>

## Changes Proposed:
-  Missing RP event after completing the quest "Deciphering the Book"

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/16225
- Closes https://github.com/chromiecraft/chromiecraft/issues/5419

## SOURCE:

https://wowpedia.fandom.com/wiki/Deciphering_the_Book
https://github.com/TrinityCore/TrinityCore/issues/18571

## Tests Performed:

- Local
- 

## How to Test the Changes:

1. .go cr 62870
2. .quest add 9557
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: data/sql/updates/pending_db_world/quest9557.sql [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+-- [Q] Deciphering the Book -- http://wotlk.openwow.com/quest=9557
+-- Anchorite Paetheus SAI
+SET @ENTRY := 17424;
+UPDATE `creature_template` SET `AIName`="SmartAI" WHERE `entry`=@ENTRY;
+DELETE FROM `smart_scripts` WHERE `entryorguid`=@ENTRY AND `source_type`=0;
+INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_x`,`target_y`,`target_z`,`target_o`,`comment`) VALUES
+(@ENTRY,0,0,0,20,0,100,0,9557,0,0,0,80,@ENTRY*100+00,2,0,0,0,0,1,0,0,0,0,0,0,0,"Anchorite Paetheus - On Quest 'Deciphering the Book' Finished - Run Script");
+
+-- Actionlist SAI
+SET @ENTRY := 1742400;
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/quest9557.sql:11]: don't reuse the same var. bad habit

Comment (Kitzunu) [reply data/sql/updates/pending_db_world/quest9557.sql:11]: you can just use @ENTRY*100

File: data/sql/updates/pending_db_world/quest9557.sql [hunk 2]
```diff
@@ -0,0 +1,19 @@
+--
+-- [Q] Deciphering the Book -- http://wotlk.openwow.com/quest=9557
+-- Anchorite Paetheus SAI
+SET @ENTRY := 17424;
+UPDATE `creature_template` SET `AIName`="SmartAI" WHERE `entry`=@ENTRY;
+DELETE FROM `smart_scripts` WHERE `entryorguid`=@ENTRY AND `source_type`=0;
+INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_x`,`target_y`,`target_z`,`target_o`,`comment`) VALUES
+(@ENTRY,0,0,0,20,0,100,0,9557,0,0,0,80,@ENTRY*100+00,2,0,0,0,0,1,0,0,0,0,0,0,0,"Anchorite Paetheus - On Quest 'Deciphering the Book' Finished - Run Script");
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/quest9557.sql:8]: `@ENTRY*100+00` this makes no sense

Comment (Kitzunu) [reply data/sql/updates/pending_db_world/quest9557.sql:8]: remove the +00

File: data/sql/updates/pending_db_world/quest9557.sql [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+-- [Q] Deciphering the Book -- http://wotlk.openwow.com/quest=9557
+-- Anchorite Paetheus SAI
+SET @ENTRY := 17424;
+UPDATE `creature_template` SET `AIName`="SmartAI" WHERE `entry`=@ENTRY;
+DELETE FROM `smart_scripts` WHERE `entryorguid`=@ENTRY AND `source_type`=0;
+INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_x`,`target_y`,`target_z`,`target_o`,`comment`) VALUES
+(@ENTRY,0,0,0,20,0,100,0,9557,0,0,0,80,@ENTRY*100,2,0,0,0,0,1,0,0,0,0,0,0,0,"Anchorite Paetheus - On Quest 'Deciphering the Book' Finished - Run Script");
+
+-- Actionlist SAI
+DELETE FROM `smart_scripts` WHERE `entryorguid`=@ENTRY AND `source_type`=9;
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/quest9557.sql:11]:
```suggestion
DELETE FROM `smart_scripts` WHERE `entryorguid`=@ENTRY*100 AND `source_type`=9;
```

File: data/sql/updates/pending_db_world/quest9557.sql [hunk 4]
```diff
@@ -0,0 +1,18 @@
+--
+-- [Q] Deciphering the Book -- http://wotlk.openwow.com/quest=9557
+-- Anchorite Paetheus SAI
+SET @ENTRY := 17424;
+UPDATE `creature_template` SET `AIName`="SmartAI" WHERE `entry`=@ENTRY;
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/quest9557.sql:5]:
```suggestion
UPDATE `creature_template` SET `AIName`='SmartAI' WHERE `entry`=@ENTRY;
```

**Review by Gultask — COMMENTED** [review]
Change double quotes to single quotes

**Comment by Gultask** [comment]
Add Rushor as a Co-Author

**Comment by Gultask** [comment]
@Tony931023 for this and the other one, try linking the commit in a comment ( -- ) and then credit Rushor in the commit description.
If you use GithubDesktop, it's here:
image

**Comment by Tony931023** [comment]
> @Tony931023 for this and the other one, try linking the commit in a comment ( -- ) and then credit Rushor in the commit description. If you use GithubDesktop, it's here: image

Co-Authored-By: Rushor <Rushor@users.noreply.github.com>

**Comment by elthehablo** [comment]
> > @Tony931023 for this and the other one, try linking the commit in a comment ( -- ) and then credit Rushor in the commit description. If you use GithubDesktop, it's here: image
> 
> Co-Authored-By: Rushor [Rushor@users.noreply.github.com](mailto:Rushor@users.noreply.github.com)

not in the comments. in the commit message, my friend :) see

`$ git commit -m "Refactor usability tests.`
`>`
`>`
`Co-authored-by: NAME <NAME@EXAMPLE.COM>`
`Co-authored-by: ANOTHER-NAME <ANOTHER-NAME@EXAMPLE.COM>"`

File: data/sql/updates/pending_db_world/quest9557.sql [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_x`,`target_y`,`target_z`,`target_o`,`comment`) VALUES
+(@ENTRY*100,9,0,0,0,0,100,0,0,0,0,0,83,2,0,0,0,0,0,1,0,0,0,0,0,0,0,'Anchorite Paetheus - On Script - Remove Npc Flag Questgiver'),
+(@ENTRY*100,9,1,0,0,0,100,0,1000,1000,0,0,1,0,4000,0,0,0,0,1,0,0,0,0,0,0,0,'Anchorite Paetheus - On Script - Say Line 0'),
+(@ENTRY*100,9,2,0,0,0,100,0,4000,4000,0,0,1,1,6000,0,0,0,0,1,0,0,0,0,0,0,0,'Anchorite Paetheus - On Script - Say Line 1'),
+(@ENTRY*100,9,3,0,0,0,100,0,6000,6000,0,0,1,2,6000,0,0,0,0,1,0,0,0,0,0,0,0,'Anchorite Paetheus - On Script - Say Line 2'),
+(@ENTRY*100,9,4,0,0,0,100,0,6000,6000,0,0,1,3,0,0,0,0,0,1,0,0,0,0,0,0,0,'Anchorite Paetheus - On Script - Say Line 3'),
+(@ENTRY*100,9,5,0,0,0,100,0,0,0,0,0,82,2,0,0,0,0,0,1,0,0,0,0,0,0,0,'Anchorite Paetheus - On Script - Add Npc Flag Questgiver');
```
Comment (Gultask) [inline data/sql/updates/pending_db_world/quest9557.sql:17]:
```suggestion
(@ENTRY*100,9,5,0,0,0,100,0,0,0,0,0,82,2,0,0,0,0,0,1,0,0,0,0,0,0,0,'Anchorite Paetheus - On Script - Add Npc Flag Questgiver');

```

**Review by Gultask — APPROVED** [review]

**Pull request merged**